ngboost/distns/betabernoulli.py

```diff
@@ -0,0 +1,125 @@
+from scipy.stats import betabinom as dist
+from scipy.stats import beta as betadist
+import numpy as np
+from ngboost.distns.distn import RegressionDistn
+from ngboost.scores import LogScore
+from scipy.special import polygamma, gamma, digamma
+from scipy.special import beta as betafunction
+from fastbetabino import *
+from array import array  
+import sys   
+class BetaBernoulliLogScore(LogScore): 
+    
+    def score(self, Y):
+        return -self.dist.logpmf(Y)
+
+    def d_score(self, Y):
+        D = np.zeros((len(Y), 2)) # first col is dS/d(log(α)), second col is dS/d(log(β))
+
+        D[:, 0] =   -self.alpha * (
+                        digamma(self.alpha + self.beta) + 
+                        digamma(Y + self.alpha) -
+                        digamma(self.alpha + self.beta + 1) -
+                        digamma(self.alpha)
+                    )
+        D[:, 1] =   -self.beta * (
+                        digamma(self.alpha + self.beta) + 
+                        digamma(-Y + self.beta + 1) -
+                        digamma(self.alpha + self.beta + 1) -
+                        digamma(self.beta)
+                    )
+        return D
+    
+    def metric(self):
```

Comment:
How did you derive this? In my derivation the Fisher Information is not diagonal.

Here's what I have; it'd be great if you could paste your independent derivation so that we can double check.

![Screen Shot 2020-06-19 at 4 04 26 PM](https://user-images.githubusercontent.com/1163598/85184902-8d854e00-b246-11ea-84de-3e599eb8ba4a.png)

Reply:
I made a mistake not including the other diagonal. My calculation was based on the definition of the FI matrix as the variance of the score. Therefore I just simply squared the gradient (but forgot that it's actually a vector and the square should be S*S.T). 
Can we use the double derivative? Are we using this:
_Claim: The negative expected Hessian of log likelihood is equal to the Fisher Information Matrix F._
[https://wiseodd.github.io/techblog/2018/03/11/fisher-information/](https://wiseodd.github.io/techblog/2018/03/11/fisher-information/)

Reply:
As per my calculation the last row is different, it's
![formula](https://latex.codecogs.com/gif.latex?\frac{d^2}{db^2}\log&space;p(x)=\frac{d}{db}\log&space;p(x)&plus;\beta^2(\phi(-x&plus;\beta&plus;1)&plus;\phi(\alpha&plus;\beta)-\phi(\alpha&plus;\beta&plus;1)-\phi(\beta)))
However when I use that formula the model doesn't work - it drops the singular matrix error again. And when I include the full FI matrix from the variance definition it also drops the singular matrix error. When I use the diagonal matrix from the Hessian definition it simply doesn't learn. 
So the only working solution is the diagonal matrix from the variance definition. I have no clue why.
If you want to try the different approaches I have updated the code. All you need to do is to comment out the other diagonal or to comment out the other metric definition. For now I keep the working version as my pull request, but please check my code as I'm not sure I didn't miss something or didn't do a typo again.